packages/react-native/React/Views/RCTView.m

```diff
@@ -923,26 +923,40 @@ static void RCTUpdateHoverStyleForView(RCTView *view)
 
 - (void)updateClippingForLayer:(CALayer *)layer
 {
-  CALayer *mask = nil;
-  CGFloat cornerRadius = 0;
-
-  if (self.clipsToBounds) {
-    const RCTCornerRadii cornerRadii = [self cornerRadii];
-    if (RCTCornerRadiiAreEqual(cornerRadii)) {
-      cornerRadius = cornerRadii.topLeft;
-
-    } else {
-      CAShapeLayer *shapeLayer = [CAShapeLayer layer];
-      CGPathRef path =
-          RCTPathCreateWithRoundedRect(self.bounds, RCTGetCornerInsets(cornerRadii, UIEdgeInsetsZero), NULL);
-      shapeLayer.path = path;
-      CGPathRelease(path);
-      mask = shapeLayer;
+    CALayer *mask = nil;
+    CGFloat cornerRadius = 0;
+    
+    if (self.clipsToBounds) {
+        const RCTCornerRadii cornerRadii = [self cornerRadii];
+        if (RCTCornerRadiiAreEqual(cornerRadii)) {
+            cornerRadius = cornerRadii.topLeft;
+            
+        } else {
+            RCTCornerInsets cornerInsets = RCTGetCornerInsets(cornerRadii, UIEdgeInsetsZero);
+            mask = RCTCreateMaskLayer(self.bounds, cornerInsets);
+        }
+    }
+    
+    layer.cornerRadius = cornerRadius;
+    layer.mask = mask;
+    
+    for (UIView *subview in self.subviews) {
+        if ([subview isKindOfClass:[UIImageView class]]) {
```

Comment:
There has been a suggestion from @NickGerleman to try and look for a more general approach.
It's not just images that should present this behavior, but all the components that have as tyle with `overflow: hidden`. Images have that style by default.

Do you think we can find a way to check if the style applied has the `overflow` property set to `hidden`, rather than checking if the subview is an Image?

Reply:
Seems like the overflow property is used [here](https://github.com/facebook/react-native/blob/main/packages/react-native/React/Views/RCTViewManager.m#L206-L213) to set `clipsToBounds`.